pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -209,6 +250,28 @@ func (c *Reconciler) createBuild(tr *v1alpha1.TaskRun) (*buildv1alpha1.Build, er
 		return nil, fmt.Errorf("Task %s has nil BuildSpec", t.Name)
 	}
 
+	bSpec := t.Spec.BuildSpec.DeepCopy()
+	bSpec.Volumes = append(bSpec.Volumes, corev1.Volume{
+		Name: resources.MountName,
+		VolumeSource: corev1.VolumeSource{
+			PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+				ClaimName: pvcName,
+			},
+		},
+	})
+
+	// Override the entrypoint so that we can use our custom
+	// entrypoint which copies logs
+	err = resources.AddEntrypoint(bSpec.Steps)
```

Comment:
Can we change the [Concepts.doc](https://github.com/knative/build-pipeline/blob/master/Concepts.md#task) to mention we no longer depend on a `BuilderImage` ?
You can also add a note for it and then we can decide what this image should be.

Reply:
I'm not sure I understand, I can update the doc to explain the entrypoint log change but we still depend on a BuilderImage if my understanding is correct

Reply:
I've update this doc with entrypoint info

Reply:
I'm guessing @tejal29 is talking about the fact that [the builder image contract]( https://github.com/knative/docs/blob/master/build/builder-contract.md#atypical-builders) actually says to do the _opposite_ of what we want - it says we want ppl to define an entrypoint when in fact we definitely do _not_ want that

Reply:
I created some docs but I'm having a hard time adding them to this PR

Reply:
I'm failing at this, I'll just make a new PR with the doc changes

Reply:
#184